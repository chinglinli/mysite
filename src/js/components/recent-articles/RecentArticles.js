import * as React from 'react';
import ImmutablePropTypes from 'react-immutable-proptypes';
import {Link} from 'react-router';

import styles from './RecentArticles.scss';


export default class RecentArticles extends React.Component {
  static PropTypes = {
    articles: ImmutablePropTypes.listOf(
      ImmutablePropTypes.contains({
        slug: React.PropTypes.string,
        title: React.PropTypes.string
      })
    )
  };

  render() {
    const {articles} = this.props;

    return (
      <div className={styles.root}>
        <div className={styles.title}>最新文章</div>
        <div className={styles.hr}/>
        <div className={styles.articles}>
          {
            articles.map(article => (
              <Link key={article.get('slug')}
                    className={styles.article}
                    to={`/articles/${article.get('slug')}/`}>
                {article.get('title')}
              </Link>
            ))
          }
        </div>
        <div className={styles.hr}/>
      </div>
    );
  }
}
