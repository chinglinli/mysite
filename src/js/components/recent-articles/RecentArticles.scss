@import '../../../css/base.scss';

.root {
  margin-top: 30px;

  .title {
    color: #6f6f6f;
    font-size: 1.5em;
    font-weight: bold;
  }

  .hr {
    margin-top: 10px;
    border-bottom: 1px solid #ddd;
  }

  .articles {
    margin-top: 1em;
    .article {
      @extend .link;
      display: block;
      &:not(first-child) {
        margin-top: .8em;
      }
    }
  }
}

