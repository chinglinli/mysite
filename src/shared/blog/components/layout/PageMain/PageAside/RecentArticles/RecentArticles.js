import React from 'react'
import PropTypes from 'prop-types'
import ImmutablePropTypes from 'react-immutable-proptypes'
import styled from 'styled-components'

import NormalLink from '../../../../generic/NormalLink'

const Base = styled.section`
  margin: 30px;
  width: 340px;
  
  @media (max-width: 1200px) {
    display: none;
  }

  h2 {
    color: #6F6F6F;
    font-size: 1.5em;
    font-weight: 700;
    border-bottom: 1px solid #DDDDDD;
    padding-bottom: 10px;
  }

  li:not(:first-child) {
    margin-top: 8px;
  }
`

export default class RecentArticles extends React.Component {
  static PropTypes = {
    recentArticles: ImmutablePropTypes.listOf(
      ImmutablePropTypes.contains({
        slug: PropTypes.string,
        title: PropTypes.string
      })
    )
  }

  render () {
    return (
      <Base>
        <h2>近期文章</h2>
        <ul>
          {
            this.props.recentArticles.map(article => (
              <li key={article.get('slug')}>
                <NormalLink to={`/articles/${article.get('slug')}/`}>
                  {article.get('title')}
                </NormalLink>
              </li>
            ))
          }
        </ul>
      </Base>
    )
  }
}
