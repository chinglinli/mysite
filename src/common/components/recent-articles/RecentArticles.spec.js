import Immutable from 'immutable'
import React from 'react'
import renderer from 'react-test-renderer'

import RecentArticles from './RecentArticles'

test('It should render RecentArticles component with empty articles correctly', () => {
  const articles = Immutable.fromJS([])

  const component = renderer.create(
    <RecentArticles articles={articles}/>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('It should render RecentArticles component correctly', () => {
  const articles = Immutable.fromJS([
    {slug: 'slug 1', title: 'title 1'},
    {slug: 'slug 2', title: 'title 2'},
    {slug: 'slug 3', title: 'title 3'}
  ])

  const component = renderer.create(
    <RecentArticles articles={articles}/>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
