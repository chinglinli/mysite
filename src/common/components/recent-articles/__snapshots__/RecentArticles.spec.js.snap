// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should render RecentArticles component correctly 1`] = `
<div
  className="root"
>
  <div
    className="title"
  >
    近期文章
  </div>
  <div
    className="hr"
  />
  <div
    className="articles"
  >
    <a
      className="article"
      onClick={[Function]}
      style={Object {}}
    >
      title 1
    </a>
    <a
      className="article"
      onClick={[Function]}
      style={Object {}}
    >
      title 2
    </a>
    <a
      className="article"
      onClick={[Function]}
      style={Object {}}
    >
      title 3
    </a>
  </div>
  <div
    className="hr"
  />
</div>
`;

exports[`It should render RecentArticles component with empty articles correctly 1`] = `
<div
  className="root"
>
  <div
    className="title"
  >
    近期文章
  </div>
  <div
    className="hr"
  />
  <div
    className="articles"
  />
  <div
    className="hr"
  />
</div>
`;
